force-app/main/default/classes/SOQL.cls

```diff
@@ -312,50 +385,89 @@ public with sharing class SOQL {
+                    OpportunityId = cloneOppId,
+                    PricebookEntryId = origLineItem.PricebookEntryId,
+                    UnitPrice = origLineItem.UnitPrice,
+                    Discount = origLineItem.Discount,
+                    Description = origLineItem.Description
+                );
+                
+                // Generate a random number between 1 and 3                
+                Double randomQuantityIncrease = Math.floor(Math.random() * (max - min + 1)) + min;   
+                // Add the random number to the quantity
+                newLineItem.Quantity = origLineItem.Quantity + randomQuantityIncrease;
+    
+                // Add the new line item to the list
+                newLineItems.add(newLineItem);
+                    }
+        }
+    
         // Insert the new line items
+        if (!newLineItems.isEmpty()) {
+            insert newLineItems;
+        }
     }
+
     
     /**
      * Question 24
      * Retrieves the Opportunity Name, Account Name, and Owner Name for Opportunities that have a Stage of 'Closed Won'.
      * Example: getOpportunityDetailsForClosedWon()
      * @return A list of filtered Opportunity
      */
-    public static List<Opportunity> getOpportunityDetailsForClosedWon() {        
-        return null; // Replace null with the variable you used to store the result
+    public static List<Opportunity> getOpportunityDetailsForClosedWon() {
+        
+        List<Opportunity> closedWonOpportunities = [SELECT Name, Account.Name, Owner.Name FROM Opportunity WHERE StageName = 'Closed Won'];
+        return closedWonOpportunities; // Replace null with the variable you used to store the result
     }
-
+    List<Opportunity> closedWonOpps = getOpportunityDetailsForClosedWon();
     /**
      * Question 25
      * Sets the 'Primary_Contact__c' field for all Closed Won Opportunities to first Contact on the Account.
      * Example: setPrimaryContact() should set the 'Primary_Contact__c' field for every Opportunity in the org
-     */
+    */
     public static void setPrimaryContact() {
         //Use the getOpportunityDetailsForClosedWon() method to get the Opportunities
-
-        // Create a set of Account Ids from the Opportunities
-
-        // Query the Contacts
-
-        // Loop through each Contact and add the Account Id and Contact to the map
+        List<Opportunity> closedWonOpps = getOpportunityDetailsForClosedWon();
         
+        // Create a set of Account Ids from the Opportunities
+        Set<Id> accountIds = new Set<Id>();
+        for (Opportunity opp : closedWonOpps) {
+            accountIds.add(opp.AccountId);
+        }
+        // Query the Contacts       
+        Map<Id, Contact> contactsMap = new Map<Id, Contact>();
         // Loop through each Opportunity
+        List<Account> getContacts = [SELECT Id, (SELECT ID, Name, AccountId, FirstName FROM Contacts ORDER BY FirstName) FROM Account WHERE ID IN :accountIds];
```

Comment:
I would rename this variable to accountList or accountsWithContacts ... something like that